packages/realm/experimental/base-url.js

```diff
@@ -0,0 +1,24 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// This file is needed as a workaround for Metro unexpectedly failing
+// to use the entry point defined in the package.json `exports` field,
+// despite enabling `unstable_enablePackageExports`.
```

Comment:
Well - even if this wasn't a problem, we would probably need this file anyway, to workaround the fact that `unstable_enablePackageExports` is disabled by default 🤔

Reply:
Also ... I know for a fact that the Metro team would be very grateful if _someone_ would set up some tests of the metro resolver (most likely copying in tests from other packages / tools). This would make a good test case if the pattern isn't already covered by one of those other tests.

Reply:
I've [updated](https://github.com/realm/realm-js/pull/6518/commits/c3064c8aeff6dcbe5656afcbaf346b96ed30dfed) the comment to incorporate being disabled by default. Feel free to suggest any additional changes you think would be relevant for this file 👍 

I'd be glad to look over some of the Metro tests and see what could be added. It does seem like their tests are covering [subpath exports](https://github.com/facebook/metro/blob/ce266ddafd5eb484168f973f84a203d8b162f842/packages/metro-resolver/src/__tests__/package-exports-test.js#L235).